python/cudf/cudf/core/column/column.py

```diff
@@ -649,14 +650,28 @@ def append(self, other):
         return ColumnBase._concat([self, as_column(other)])
 
     def quantile(self, q, interpolation, exact):
-        if isinstance(q, Number):
+
+        is_number = isinstance(q, Number)
+
+        if is_number:
             quant = [float(q)]
         elif isinstance(q, list) or isinstance(q, np.ndarray):
             quant = q
         else:
             msg = "`q` must be either a single element, list or numpy array"
             raise TypeError(msg)
-        return libcudf.quantile.quantile(self, quant, interpolation, exact)
+
+        # get sorted indicies and exclude nulls
+        sorted_indices = self.as_frame().get_sorted_inds(True, "after")
+        sorted_indices = sorted_indices[self.null_count :]
+
+        return cpp_quantile(
+            self,
+            quant,
+            Interpolation[interpolation.upper()],
```

Comment:
Can we move this logic down to the Cython and just pass the string `interpolation` as-is to `cpp_quantile`?

Reply:
Yes.

Do we / should we have a doc indicating guidance with regard to the cython layer? Would be nice to reference that ahead of time. :)

Reply:
I started work on something like that in https://github.com/rapidsai/cudf/pull/3918 but it lost steam :(

Maybe we can prioritize something like it for 0.14.